xmake/modules/package/manager/pacman/find_package.lua

```diff
@@ -65,28 +142,54 @@ function main(name, opt)
             has_includes = true
         end
     end
-
-    -- get pkgconfig file
-    local pkgconfig_file = pkgconfig_files[name]
-    if not pkgconfig_file then
-        for _, file in pairs(pkgconfig_files) do
-            pkgconfig_file = file
-            break
-        end
-    end
-
-    -- find package
-    local result = nil
-    if pkgconfig_file then
+    
+    -- we iterate over each pkgconfig file to extract the required data
+    local result = {
+        includedirs = {},
+        linkdirs = {},
+        links = {},
+        version = nil
+    }
+    
+    local foundPC = false
+    
+    for key, file in pairs(pkgconfig_files) do
+        pkgconfig_file = file
         local pkgconfig_dir = path.directory(pkgconfig_file)
         local pkgconfig_name = path.basename(pkgconfig_file)
         linkdirs = table.unique(linkdirs)
         includedirs = table.unique(includedirs)
-        result = find_package_from_pkgconfig(pkgconfig_name, {configdirs = pkgconfig_dir, linkdirs = linkdirs})
-        if not result and has_includes then
-            -- header only and hidden /usr/include? we need only return empty {}
-            result = {}
+        local pcresult = {}
+        pcresult = find_package_from_pkgconfig(pkgconfig_name, {configdirs = pkgconfig_dir, linkdirs = linkdirs})
+            
+        -- the pkgconfig file has been parse successfully
+        if pcresult ~= nil then
+            for _, locincludedir in pairs(pcresult.includedirs) do
+                table.insert(result.includedirs, locincludedir)
+            end
+                    
+            for _, loclinkdir in pairs(pcresult.linkdirs) do
+                table.insert(result.linkdirs, loclinkdir)
+            end
+            
+            for _, loclink in pairs(pcresult.links) do
+                table.insert(result.links, loclink)
+            end
+
+            -- version should be the same if a pacman package contains multiples .pc
+            result.version = pcresult.version
+            
+            foundPC = true
         end
     end
+        
+    if foundPC == true then
+        result.includedirs = table.unique(result.includedirs)
+        result.linkdirs = table.unique(result.linkdirs)
+        result.links = table.unique(result.links)
+    else -- if there is no .pc, we parse the package content to obtain the data we want
```

Comment:
you need add `elseif cygpath then` to ensure it is found

In addition, you need to be compatible with archlinux/pacman, currently I see it only supports msys/pacman

Reply:
Normally, I only use cypath inside a `if is_subhost("msys") and opt.plat == "mingw" then end` to modify the obtained path in the case I am under msys. So the code should work on archlinux, but I do not have archlinux to test it.

Reply:
I will test it.